src/App.tsx

```diff
@@ -14,11 +14,130 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  goods: string[];
+  isVisible: boolean;
+  isReversed: boolean;
+  sortBy: 'name' | 'length' | null;
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    isVisible: false,
+    isReversed: false,
+    sortBy: null,
+  };
+
+  start = () => {
+    this.setState({ isVisible: true });
+  };
+
+  reverse = () => {
+    this.setState((state) => (
+      { isReversed: !state.isReversed }
+    ));
+  };
+
+  sortAlphabetically = () => {
+    this.setState({
+      sortBy: 'name',
+      isReversed: false,
+    });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: null,
+    });
+  };
+
+  render() {
+    const goodsList: string[] = [...this.state.goods];
+
+    switch (this.state.sortBy) {
+      case 'name':
+        goodsList.sort((first, second) => first.localeCompare(second));
+        break;
+      case 'length':
+        goodsList.sort((first, second) => second.length
+        - first.length);
```

Comment:
second.length - first.length makes it sorted descending. If `sort` i think it should be ascending.